test/plugin_functional/test_suites/core_plugins/rendering.ts

```diff
@@ -261,6 +261,9 @@ export default function ({ getService }: PluginFunctionalProviderContext) {
         'xpack.discoverEnhanced.actions.exploreDataInContextMenu.enabled (boolean)',
         'xpack.fleet.agents.enabled (boolean)',
         'xpack.fleet.agentless.api.url (string)',
+        'xpack.fleet.agentless.api.tls.certificate (string)',
+        'xpack.fleet.agentless.api.tls.key (string)',
```

Comment:
@kc13greiner I hope this answers your questions and concerns.

They are secrets and they will be used by the [Fleet plugin's server code ](https://github.com/elastic/kibana/pull/187488/files#diff-11bcc5bb1eaf5ddae753f9ee690233bef988d13ddbf6baf71c4c8edd0d6a4933R1110)to make calls to the Agentless API in the ESS's new MKI.   They will not be used on the client-side code of the plugin.

They will likely not be new secrets,  AFAIK now, we will be using the same certificate that the synthetic plugin 

We are using the precedent [here](https://github.com/elastic/cloud/pull/95629) and [implemented here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/synthetics/server/synthetics_service/service_api_client.ts#L138) by the Observability Synthetics team.

Eventually,  we will be configuring this in the cloud and cloud_assets repo to be configured during the deployment of an ESS environment.

Reply:
@kc13greiner,  does adding the configs to this imply that they are client-facing UI configs?

Reply:
@seanrathier 

> They are secrets and they will be used by the [Fleet plugin's server code ](https://github.com/elastic/kibana/pull/187488/files#diff-11bcc5bb1eaf5ddae753f9ee690233bef988d13ddbf6baf71c4c8edd0d6a4933R1110)to make calls to the Agentless API in the ESS's new MKI. They will not be used on the client-side code of the plugin.

Thanks for the information. Since these are secrets and not needed by the UI, we should ensure they are not being exposed to the UI.

> does adding the configs to this imply that they are client-facing UI configs?
No. `test/plugin_functional/test_suites/core_plugins/rendering.ts` is a test file that catches new values being exposed to the UI so that we can review them as CodeOwners.

So we have to figure out why they are showing up in this test file.

I believe that because in `x-pack/plugins/fleet/server/config.ts` on L30:
```js
export const config: PluginConfigDescriptor = {
  exposeToBrowser: {
    epm: true,
    agents: {
      enabled: true,
    },
    agentless: true,
    enableExperimental: true,
    developer: {
      maxAgentPoliciesWithInactivityTimeout: true,
    },
    internal: {
      fleetServerStandalone: true,
      activeAgentsSoftLimit: true,
      onlyAllowAgentUpgradeToKnownVersions: true,
    },
  }
```
`agentless: true` is exposing all sub-configs to be exposed.

You may want to add a more explicit definition of what under `agentless` should be exposed. Or add these new config options under a different top-level config.

Reply:
Ok, I think I understand now.  In the `rendering.ts` and `exposeToBrowser` we should only have what config the browser will use.  In this case, `agentless.api.url` we want it exposed so the it should be like this

```
exposeToBrowser: {
    epm: true,
    agents: {
      enabled: true,
    },
    agentless: {
      api: {
        url: true,
      },
    },
    enableExperimental: true,
    developer: {
      maxAgentPoliciesWithInactivityTimeout: true,
    },
    internal: {
      fleetServerStandalone: true,
      activeAgentsSoftLimit: true,
      onlyAllowAgentUpgradeToKnownVersions: true,
    },
  },
```

Reply:
Correct; `rendering.ts` asserts all values exposedToBrowser as a test.

As long as `agent.api.url` is not a secret and there weren't any other sub-configs of `agentless` that also needed to be exposed, that structure you posted would be good to go.